compiler/sem/sem.nim

```diff
@@ -350,18 +350,14 @@ proc newSymG*(kind: TSymKind, n: PNode, c: PContext): PSym =
   of nkSym:
     # and sfGenSym in n.sym.flags:
     result = n.sym
-    if result.kind notin {kind, skTemp}:
-      localReport(c.config, n.info, SemReport(
-        kind: rsemSymbolKindMismatch,
-        sym: result,
-        expectedSymbolKind: {kind}))
-
-    when false:
-      if sfGenSym in result.flags and result.kind notin {skTemplate, skMacro, skParam}:
-        # declarative context, so produce a fresh gensym:
-        result = copySym(result)
-        result.ast = n.sym.ast
-        put(c.p, n.sym, result)
+    doAssert n.sym.kind in {kind, skTemp},
+      "there is a bug the callee or sanitizer, got: $#, expected: $#" %
+        [$n.sym.kind, $kind]
```

Comment:
It's definitely possible to leave the symbol check to the sanitizer, although I'm still a bit undecided on whether the check is something that fits better here.

In its current form, the assert would also conflict with the [tests/lang_callable/macros/tsymbol_must_match.nim](https://github.com/nim-works/nimskull/pull/590/commits/f22e1b69540ac2db7b61ed4c2683c50058ebf8a8#diff-128b474d331f2e1dcb7bfc6a78d5ece9433bcc828ed2c0559880cb26c7fe19bc) test added by #590.

Anyhow, I don't think the assertion is a blocker, and am okay with the removing the test mentioned above again.

Reply:
Thanks for the review. I'm ok with dropping the assertion, mostly I think the error generation shouldn't happen in this proc and we should have prevented/figured things out before hand. Where that figuring out happens very much depends upon how we got here.